js/store/actions.js

```diff
@@ -1,11 +1,42 @@
 import fetchItems from '../api/fetch-items';
+import { geoWatch } from '../api/geolocation';
 
 import {
   EMPTY,
   FETCHED,
   ERRORED,
 } from './constants';
 
+export const watchGeoLocation = () => {
+  return (dispatch) => {
+    dispatch(geoAdquiring);
+    geoWatch((geo) => {
+      dispatch(geoAdquired(geo));
+    },() => {
+      dispatch(geoError);
+    });
+  };
+};
+
+export const geoAdquiring = () => {
+  return {
+    type: 'GEO_ADQUIRING',
+  };
+};
+
+export const geoAdquired = (value) => {
+  return {
+    type: 'GEO_ADQUIRED',
+    value,
+  };
+};
+
+export const geoError = () => {
+  return {
+    type: 'GEO_ERROR',
```

Comment:
Use past-tense for actions, something that has happened and we are responding to.